Use devtools

MartinSchobben opened the pull request:
I have made some changes here to the Dockerfile to enable installing Devtools. The shell script now helps me to better evaluate the package when build against irods. I can create documentation now, but installing the package (`devtools::build()`) produces an error. It says `-stdlib=libc++` should not be included in the Makevars file.

korydraughn committed:
It compiles against 4.3.0!

trel committed:
docker builder WIP

trel committed:
getting closer

trel committed:
more pandoc

MartinSchobben committed:
ignore top level non-standard files

MartinSchobben committed:
Merge branch 'docker-on-kory' into use_devtools

MartinSchobben committed:
additions to buildignore

MartinSchobben committed:
updates to Dockefile and a shell script for devtools building and documenting

MartinSchobben committed:
shell script

trel commented on build-rirods.sh:
```diff
@@ -0,0 +1,24 @@
+sudo docker image build --tag rirods:build .
+
+# local source and library paths
+SRCPATH=$(R -e "getwd()"| grep /)
+LIBPATH=$(R -e ".libPaths()[1]" | grep /)
```
this requires R is installed on the machine running docker... I was hoping to have little/no dependencies on the host machine.

MartinSchobben replied:
My main idea was to help the development of the R package by adding this file (can be deleted later-on). So, for that reason, I use volumes to execute R code (devtools) in the container but have my files updated. This file is also included in the buildignore.

trel replied:
Got it - I see.

trel commented on R/RcppExports.R:
```diff
@@ -0,0 +1,276 @@
+# Generated by using Rcpp::compileAttributes() -> do not edit by hand
```
if this is a generated file, can we ... not have it in the source?

MartinSchobben replied:
I would argue that it needs to be included. Then use wrapper function around these automatic generated functions to add the custom R code (which is now included in RccpExportsManual.R) and divide over several R scripts with some logical division of functionality. This is then just for clarity of the code.

trel commented on NAMESPACE:
```diff
@@ -1,3 +1,2 @@
-#useDynLib(rirods)
```
if this is a generated file, can we ... not have it in the source?

trel commented on build-rirods.sh:
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# build image
+sudo docker image build --tag rirods:build .
```
shouldn't need sudo in here...

trel commented on build-rirods.sh:
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
might rework this to be a Makefile itself...

targets could include...
- dockerimage
- devtoolsdocument
- devtoolsbuild

MartinSchobben replied:
Custom makefiles are advised against (https://r-pkgs.org/src.html?q=Make#make). I deleted the original Makefile in this branch.

trel replied:
Yes, I was talking about a Makefile to provide the same functionality as this shell script - just calling docker itself.

No worries, now I read the interaction could be messy.

trel reviewed (commented):
will run this later today - wrote some initial comments via only inspection, ignore as appropriate :)

MartinSchobben committed:
install latest r version, correct usage of namespace

MartinSchobben committed:
use rocker

MartinSchobben committed:
open R project and load devtools

MartinSchobben committed:
version roxygen and some other small fixes

MartinSchobben committed:
add system requirements to description and readme files

trel committed:
fix syntax error, only one creator

trel committed:
update github location

trel committed:
Rf_error shenanigans

trel committed:
NO DYNAMIC LIBRARY!  NO!

MartinSchobben committed:
build rocker image

MartinSchobben committed:
correct rocker image build recipe

Merge branch 'use_devtools' of https://github.com/MartinSchobben/irods_client_library_r_cpp into use_devtools
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

trel commented:
We have abandoned this effort - and have decided to deprecate and archive this repository.

Please find a new pure-R implementation at:

https://github.com/irods/irods_client_library_rirods
